project/src/com/google/daggerquery/executor/QueryExecutor.java

```diff
@@ -49,18 +50,25 @@ public static void main(String[] args) {
       ImmutableList.Builder resultBuilder = new ImmutableList.Builder();
 
       // We assume that we successfully executed a query only if in at least one graph it was executed without fail.
-      String failureReason = "";
+      MisspelledNodeNameException lastMisspelledNodeNameException = null;
+      IllegalArgumentException lastIllegalArgumentException = null;
```

Comment:
This pattern doesn't look very sustainable as more exception types are added. How about just keeping a single variable `Exception e` if you only care about any exception?

Reply:
Yes, I'm not happy with my approach either, but keeping any `Exception` will also not work: `MisspelledNodeNameException` has a higher priority, and we want to show the user the fix, if any.

The second idea is to cast the last saved exception to  `MisspelledNodeNameException` and only rewrite it if the cast has failed. Is it better?

Reply:
I see, thanks I was missing this earlier. How about just keeping a variable denoting the last seen exception, and break out of the loop once a `MisspelledNodeNameException` has been caught?

Reply:
What if in the first graph we will catch `MisspelledNodeNameException` with a fix from `com.google.Cat` to `com.google.Can`, but in the second graph `com.google.Cat` node exists? It sounds like a very rare case, buuut 🤔

Reply:
That's true :joy: I'm not sure what would be the ideal way now - another idea I can think of is to collect all the errors in a list, then later have aggregation logic to combine these errors for display (e.g. sort the list by importance), but this sounds overkill and may not make the code any cleaner :disappointed: 
Let's do it your way until we have a better solution.